### PR TITLE
[1.x] Dynamic properties handling

### PR DESCRIPTION
**Pull request by fabio-ivona**
Implements dynamic properties handling in order to avoid deprecation in PHP 8.2

fix https://github.com/pestphp/pest/issues/555

a PR for v2 will follow this one

### CODE REVIEW

File: tests/Hooks/AfterEachTest.php [hunk 1]
```diff
@@ -1,18 +1,14 @@
 <?php
 
 uses()->afterEach(function () {
-    expect($this)
-        ->toHaveProperty('ith')
```
Comment (fabio-ivona) [inline tests/Hooks/AfterEachTest.php:5]: `toHaveProperty` expectation doesn't work for properties handled with __get method

**Comment by nunomaduro** [comment]
Can't you solve this issue using the `#[AllowDynamicProperties]` thing?

**Comment by fabio-ivona** [comment]
> Can't you solve this issue using the #[AllowDynamicProperties] thing?

just wanted to fix the issue in a definitive way, but I can do that, np

**Comment by fabio-ivona** [comment]
@nunomaduro done

**Comment by nunomaduro** [comment]
Can you port that to pest v2?

**Comment by dingo-d** [comment]
@fabio-ivona @nunomaduro this is not really a fix 😬 

You are just silencing the deprecation until PHP 9 🙈 

The underlying test case should probably have a `__set($property, $propertyValue)` method or something similar in order for the properties to be creatable on the fly. 

That way, any reference to the property inside the test case that doesn't exist on the `TestCase` class would be created on the fly.

**Comment by fabio-ivona** [comment]
As it is a deprecation, it is fine to silence it, we'll plan a clean fix for php9

**Pull request merged**